plugins/dev-create/templates/typescript-plugin/template/test/index.ts

```diff
@@ -36,8 +36,12 @@ document.addEventListener('DOMContentLoaded', function () {
   const defaultOptions = {
     toolbox: toolboxCategories,
   };
+  const rootElement = document.getElementById('root');
+  if (!rootElement) {
+    throw new Error(`div with id 'root' not found`);
+  }
   createPlayground(
-    document.getElementById('root'),
+    document.getElementById('root')!,
```

Comment:
Thanks for fixing this! Since you added a check to see if it's null, you shouldn't need the non-null assertion anymore as TS should understand it can't be null if we've reached this code.

Reply:
I thought it shouldn't be necessary as well, but when I run the code without it, I still get the same error.

And no worries about the response time; I don't even consider that late!

Reply:
You should use the `rootElement` variable you created instead of repeating the `getElementById` call. I think that should satisfy TS without the `!`

Reply:
> You should use the `rootElement` variable you created instead of repeating the `getElementById` call. I think that should satisfy TS without the `!`

@maribethb Yes, of course, I declared the variable `rootElement` to avoid repetition of the `getElementById` call, so I'm not sure why I didn't use it there. And for an arbitrary non-pure function, the function might not return `null` on the first call for a given input but it could on the next. So, by repeating the call to `document.getElementById('root')`, the compiler's static type check still wouldn't have been satisfied.

Not sure how I missed that, thanks!

Reply:
No worries, I missed it too on first read :)